File: manual/data/python/tests/test_boltons_iterutils_flatten.py
import unittest
from function_boltons_stringutils_clean import DataFlattener

class TestDataFlattener(unittest.TestCase):
    def setUp(self):
        self.function = DataFlattener()

    def test_flatten_data_valid_input(self):
        data_str = "1,2,3;4,5,6;7,8,9"
        result = self.function.flatten_data(data_str)
        expected = "1,2,3,4,5,6,7,8,9"
        self.assertEqual(result, expected)

    def test_flatten_data_empty_string(self):
        data_str = ""
        result = self.function.flatten_data(data_str)
        expected = "Error"
        self.assertEqual(result, expected)

    def test_flatten_data_invalid_input(self):
        data_str = 12345  # Invalid input type (not a string)
        result = self.function.flatten_data(data_str)
        expected = "Error"
        self.assertEqual(result, expected)

    def test_flatten_data_single_element(self):
        data_str = "1"
        result = self.function.flatten_data(data_str)
        expected = "1"
        self.assertEqual(result, expected)

    def test_flatten_data_mixed_structure(self):
        data_str = "1,2,3;4,5;6,7,8"
        result = self.function.flatten_data(data_str)
        expected = "1,2,3,4,5,6,7,8"
        self.assertEqual(result, expected)

if __name__ == "__main__":
    unittest.main()

File: manual/data/cpp/tests/test_boltons_iterutils_flatten.cpp
#include <gtest/gtest.h>
#include "../src/function_boltons_iterutils_flatten.cpp"

TEST(DataFlattenerTest, TestFlattenDataValidInput) {
    DataFlattener function;
    std::string data_str = "1,2,3;4,5,6;7,8,9";
    std::string result = function.flatten_data(data_str);
    std::string expected = "1,2,3,4,5,6,7,8,9";
    EXPECT_EQ(result, expected);
}

TEST(DataFlattenerTest, TestFlattenDataEmptyString) {
    DataFlattener function;
    std::string data_str = "";
    std::string result = function.flatten_data(data_str);
    std::string expected = "Error";
    EXPECT_EQ(result, expected);
}

TEST(DataFlattenerTest, TestFlattenDataInvalidInput) {
    DataFlattener function;
    int data = 12345; // 非字符串输入
    std::string result = function.flatten_data(data);
    std::string expected = "Error";
    EXPECT_EQ(result, expected);
}

TEST(DataFlattenerTest, TestFlattenDataSingleElement) {
    DataFlattener function;
    std::string data_str = "1";
    std::string result = function.flatten_data(data_str);
    std::string expected = "1";
    EXPECT_EQ(result, expected);
}

TEST(DataFlattenerTest, TestFlattenDataMixedStructure) {
    DataFlattener function;
    std::string data_str = "1,2,3;4,5;6,7,8";
    std::string result = function.flatten_data(data_str);
    std::string expected = "1,2,3,4,5,6,7,8";
    EXPECT_EQ(result, expected);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}